### ISSUE TITLE
[pact-jvm-consumer-junit] Allow to specify the target folder for generated pacts

### ISSUE THREAD

**Issue by thombergs** [DiUS/pact-jvm#514]
Currently, when using the `PactProviderRuleMk2` together with the `@Pact` annotation to create a pact, the pact files are always put into the folder `target/pacts`. It would be helpful to control the target folder for the pacts, for example by specifying the target folder in the constructor of `PactProviderRuleMk2`. 

This would make it more clear where the pact files come from and would allow for a more flexible integration in a build process with Gradle (since Gradle doesn't know about the target folder by default).

**Comment by uglyog** [comment]
That makes sense, I might look at adding it to the config options.

In the mean time, you can set the `pact.rootDir` system property. For Gradle, you can do

```groovy
test {
  systemProperty 'pact.rootDir', "$buildDir/pacts"
}
```

**Comment by meirka** [comment]
Hey, is there any progress on the feature?

**Comment by uglyog** [comment]
Nothing to date, I'll see about getting some time for this but it will be done for version 4.0.0.

Also, an update from @pkubowicz now allows pact-jvm to detect that it is running in Gradle and will set the build directory appropriately.

**Comment by uglyog** [comment]
Actually, this has been implemented as part of #840 and released with 3.6.2, but for JUnit 5.
